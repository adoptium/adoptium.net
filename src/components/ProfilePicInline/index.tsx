import React from "react"
import Image from "next/image"

interface ProfilePicInlineProps {
  identifier: string
  name: string
}

const ProfilePicInline: React.FC<ProfilePicInlineProps> = props => {
  return (
    <Image
      src={`/images/authors/${props.identifier}.jpg`}
      alt={props.name}
      width={30}
      height={30}
      className="ml-2 mb-0 rounded-full"
    />
  )
}

export default ProfilePicInline
