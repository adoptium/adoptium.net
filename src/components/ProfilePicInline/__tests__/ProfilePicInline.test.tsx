import React from "react"
import { render } from "@testing-library/react"
import { describe, expect, it } from "vitest"
import ProfilePicInline from "../index"

describe("ProfilePicInline component", () => {
  it("should render correctly", () => {
    const { container } = render(<ProfilePicInline identifier="pmc" />)

    expect(container).toMatchSnapshot()
  })

  it("should render correctly - no profile pic found", () => {
    const { container } = render(<ProfilePicInline identifier="foo" />)

    expect(container).toMatchSnapshot()
  })
})
