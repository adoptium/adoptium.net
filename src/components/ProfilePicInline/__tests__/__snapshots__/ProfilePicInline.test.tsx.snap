// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProfilePicInline component > should render correctly - no profile pic found 1`] = `<div />`;

exports[`ProfilePicInline component > should render correctly 1`] = `
<div>
  <img
    image="[object Object]"
    stlye="[object Object]"
    style="margin-left: 0.5rem; margin-bottom: 0px; min-width: 30px; border-radius: 100%;"
  />
</div>
`;
